tests/test_bot.py

```diff
@@ -1648,6 +1649,40 @@ async def post(*args, **kwargs):
             bot.callback_data_cache.clear_callback_data()
             bot.callback_data_cache.clear_callback_queries()
 
+    @pytest.mark.skipif(not TEST_WITH_OPT_DEPS, reason="Optional dependencies not installed")
+    async def test_http2_warning(self):
+        with pytest.raises(
```

Comment:
Using caplog would be more appropriate than pytest.raises. The warning being converted to an exception is just a side effect of your pytest settings, plus using caplog allows us to also test the stacklevel